packages/dappmanager/src/modules/ethClient/ethereumClient.ts

```diff
@@ -104,98 +110,158 @@ export class EthereumClient {
       await db.executionClientMainnet.set(execClient);
       await db.consensusClientMainnet.set(consClient);
       if (sync)
-        await this.changeEthClientSync(
+        await this.changeEthClientSync({
+          prevExecClient: currentTarget !== "remote" ? currentTarget.execClient : undefined,
           execClient,
           consClient,
           useCheckpointSync
-        );
+        });
       else
-        await this.changeEthClientNotAsync(
+        await this.changeEthClientNotAsync({
+          prevExecClient: currentTarget !== "remote" ? currentTarget.execClient : undefined,
           execClient,
           consClient,
           useCheckpointSync
-        );
+        });
     }
   }
 
   /**
-   * Handles the Ethereum client fullnode.dappnode alias for the execution client
-   * @param dnpName dnp name of the execution client to add/remove the alias from
-   * @param removeAlias if true, removes the alias, if false, adds it
-   */
-  async setDefaultEthClientFullNode({
-    dnpName,
-    removeAlias
+ * Updates the fullnode.dappnode alias when the execution client changes.
+ * - For mainnet: fullnode.dappnode
+ * - For other networks: <network>.fullnode.dappnode
+ * 
+ * @param newExecClientDnpName - New execution client to set.
+ * @param network - Network to define the proper alias.
+ */
+  async updateFullnodeAlias<T extends Network>({
+    prevExecClientDnpName,
+    newExecClientDnpName,
+    network = "mainnet" as T
   }: {
-    dnpName: ExecutionClientMainnet;
-    removeAlias: boolean;
+    prevExecClientDnpName?: ExecutionClient<T>;
+    newExecClientDnpName?: ExecutionClient<T>;
+    network?: T;
```

Comment:
why is `network` optional and default to mainnet? wouldnt it be better to make it explicitly clear what network we want and make it a mandatory parameter?

Reply:
I chose to do it in this way because "mainnet" is the actual purpose of the fullnode functionality and before we did not consider fullnode domain in other networks so it should be the main one